tt_metal/hw/ckernels/grayskull/metal/llk_api/llk_sfpu/ckernel_sfpu_recip.h

```diff
@@ -17,74 +18,16 @@ namespace ckernel
 namespace sfpu
 {
 
-template <bool save_reg, int max_iter = 3>
+template <bool save_reg = false, int max_iter = 3>
 sfpi_inline vFloat sfpu_reciprocal(const vFloat in)
 {
-    vInt orig_exp;
-
-    if constexpr (max_iter == 1) {
-        // If we are only doing one iteration of the MAD loop, then we only need to use one LREG for the MAD instructions because we have our "first guess" in a hard-coded register
-        // This allows us to avoid having to load back in the original value later on
-        orig_exp = exexp(in);
-    }
-
-    // Force sign to 1 (make number negative)
-    vFloat val = setsgn(in, 1);
-
-    val = setexp(val, 126); // Set exponent to 126 to make the number in 0.5-1
-    // Use 1.44 as first guess at x, ideal value would be 1.33, but we happen to have 1.44 available, so use that to avoid a load
-    vFloat two;
-    if (!save_reg) {
-        two = 2.0f;
-    }
-    vFloat result = vConst1p4424 * (val * vConst1p4424 + (save_reg ? 2.0f : two));
-
-    for (int s_iter = 0; s_iter < (max_iter-1); s_iter++) {
-        result = result * (val * result + (save_reg ? 2.0f : two));
-    }
-
-    vInt new_exp = exexp(result);
-    if constexpr (max_iter != 1) {
-        orig_exp = exexp(dst_reg[0]);
-    }
-
-    // "Subtract" exponents, and re-bias.
-    // Execute: -1 - exp, then exp += 127
-    new_exp -= orig_exp;
-    new_exp += 126;
-
-    v_if (new_exp < 0) {
-        // If rebiased exponent is negative, we need to saturate at 0.
-        // This means the initial number was too big so reciprocal result should be 0
-        result = 0.0F;
-        new_exp = 0;
-    }
-    v_endif;
-
-    // Set newly denormalized exponent to result exponent field
-    return setexp(result, new_exp);
+    return _sfpu_reciprocal_<save_reg, max_iter>(in);
 }
 
-template <bool APPROXIMATION_MODE, int ITERATIONS=4>
+template <bool APPROXIMATION_MODE, int ITERATIONS=4, bool SAVE_REG = false>
```

Comment:
You dont need to expose the SAVE_REG flag here, because it was already exposed in `_calculate_reciprocal_`

Reply:
As per discussion : It seems the flag for save_register in sfpu_reciprocal was added by customers/developers who wanted to use one less register, it is supposed to degrade performance but if they wanted to assume that certain registers persist between calls by doing so the option to pass it is already in the function but for compute_reciprocal function it was passed as true which I set to false for better performance, in case if any code where such assumption is made that a register should retain its value between multiple function calls if it is not used, then we can use the extra flag to allow the assumption to be valid.

Reply:
ok sounds good then, in case it was left there for legacy reasons.